diff --git a/webpack.config.js b/webpack.config.js
index 5278b9944..35725ffdf 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -32,7 +32,13 @@ const base = {
         chunkFilename: 'chunks/[name].js'
     },
     resolve: {
-        symlinks: false
+        symlinks: true
+    },
+    watchOptions: {
+        ignored: [
+            /node_modules([\\]+|\/)+(?!scratch-vm)/,
+            /\scratch-vm([\\]+|\/)node_modules/
+        ]
     },
     module: {
         rules: [{
